src/main/java/org/folio/services/ChangeEngineServiceImpl.java

```
@@ -45,76 +40,111 @@ public ChangeEngineServiceImpl(Vertx vertx, String tenantId) {
+                LOGGER.error(errorMessage);
+                future.fail(errorMessage);
               } else {
-                JsonObject recordCollection = countResult.result().getJson();
-                if (recordCollection == null || recordCollection.getInteger("totalRecords") == null) {
-                  String errorMessage = "Error during getting records for count records for job execution with id: " + job.getId();
+                future.complete(job);
+              }
+            });
+        }
+      });
+    return future;
+  }
+
+  private Future<Integer> receiveTotalRecordsCountByJobId(String jobId, OkapiConnectionParams params) {
+    Future<Integer> future = Future.future();
+    SourceStorageClient client = new SourceStorageClient(params.getOkapiUrl(), params.getTenantId(), params.getToken());
+    try {
+      client.getSourceStorageRecord(buildQueryByJobId(jobId), 0, DEF_CHUNK_NUMBER, null, countResult -> {
+        if (countResult.statusCode() != HttpStatus.HTTP_OK.toInt()) {
+          LOGGER.error("Error during requesting number of records for jobExecution with id: {}", jobId);
+          future.fail(new HttpStatusException(countResult.statusCode(), "Error during requesting number of records for jobExecution with id: " + jobId));
+        } else {
+          countResult.bodyHandler(buffer -> {
+            JsonObject recordCollection = buffer.toJsonObject();
+            if (recordCollection == null || recordCollection.getInteger("totalRecords") == null) {
+              String errorMessage = "Error during getting records for count records for job execution with id: " + jobId;
+              LOGGER.error(errorMessage);
+              future.fail(errorMessage);
+            } else {
+              future.complete(recordCollection.getInteger("totalRecords"));
+            }
+          });
+        }
+      });
+    } catch (UnsupportedEncodingException e) {
```

better to catch all Exceptions